java/bench/src/java/org/apache/orc/bench/floating/DoubleReadBenchmark.java

```diff
@@ -0,0 +1,251 @@
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations=1, time=1)
+@Measurement(iterations=1, time=1)
+@Fork(1)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class DoubleReadBenchmark {
+  private static final String DATA_DIRECTORY_ENV = "data.directory";
+  static final String DEFAULT_DATA_DIR = "bench/target/double-mini-orc/";
+  private static String dataDirectory = DEFAULT_DATA_DIR;
+  static {
+    if (System.getProperty(DATA_DIRECTORY_ENV) != null) {
+      dataDirectory = System.getProperty(DATA_DIRECTORY_ENV);
+    }
+  }
+
+  private static final int BATCH_SIZE = 1024;
+
+  public static TreeReaderFactory.TreeReader createReader(int column,
+                                                          TreeReaderFactory.Context context,
+                                                          DoubleWriteBenchmark.Algorithm algorithm
+                                                          ) throws IOException {
+    switch (algorithm) {
+      case PLAIN:
+        return new TreeReaderFactory.DoubleTreeReader(column);
+      case PLAIN_V2:
+        return new DoubleTreeReaderV2(column, null, context);
+      case FPC_V1:
+        return new DoubleTreeReaderFpcV1(column, null, context);
+      case FPC_V2:
+        return new DoubleTreeReaderFpcV2(column, null, context);
+      case FLIP:
+        return new DoubleTreeReaderFlip(column, null, context);
+      case SPLIT:
+        return new DoubleTreeReaderSplit(column, null, context);
+      default:
+        throw new IllegalArgumentException("Unknown algorithm " + algorithm);
+    }
+  }
+
+  @State(Scope.Benchmark)
+  public static class InputData {
+
+    @Param({"LIST_PRICE",
+        "DISCOUNT_AMT",
+        "IOT_METER",
+        "NYC_TAXI_DROP_LAT",
+        "NYC_TAXI_DROP_LONG",
+        "HIGGS",
+        "HEPMASS",
+        "PHONE"})
+    DoubleWriteBenchmark.DataSet dataSet;
+
+    @Param({"NONE", "ZLIB", "LZO"})
+    CompressionKind compression;
+
+    @Param({"PLAIN", "PLAIN_V2", "FPC_V1", "FPC_V2", "FLIP", "SPLIT"})
+    DoubleWriteBenchmark.Algorithm algorithm;
+
+    OrcProto.StripeFooter footer;
+    Map<StreamName, byte[]> streams = new HashMap<>();
+    CompressionCodec codec;
+
+    public InputData() {
+      // PASS
+    }
+
+    @Setup
+    public void setup() throws IOException {
+      String path = DoubleReadSetup.makeFilename(dataDirectory, algorithm,
+          dataSet, compression);
+      DataInputStream inStream = new DataInputStream(new FileInputStream(path));
+      int footerSize = inStream.readInt();
+      byte[] footerBuffer = new byte[footerSize];
+      inStream.readFully(footerBuffer);
+      footer = OrcProto.StripeFooter.parseFrom(footerBuffer);
+      streams.clear();
+      for(OrcProto.Stream stream: footer.getStreamsList()) {
+        StreamName name = new StreamName(stream.getColumn(), stream.getKind());
+        CompressionCodec codec = WriterImpl.createCodec(compression);
```

Comment:
Looks like this is using the old ZLIB codecs.

CompressionCodec codec = getCustomizedCodec(OrcProto.Stream.Kind.DATA); ?